lib/features/recommendation/interests/screens/interests_selection_screen.dart

```diff
@@ -0,0 +1,162 @@
+                            height: size.height * 0.07,
+                          ),
+                          if (organisationState is OrganisationsFetchingState)
+                            Center(
+                              child: LoadingAnimationWidget.waveDots(
+                                  color: const Color(0xFF54A1EE),
+                                  size: anchorSize * 0.2),
+                            ),
+                          if (organisationState is! OrganisationsFetchingState)
+                            Padding(
+                              padding: EdgeInsets.symmetric(
+                                  horizontal: size.width * 0.1),
+                              child: GridView.count(
+                                childAspectRatio: 2,
+                                crossAxisCount: 4,
+                                shrinkWrap: true,
+                                children: state.interests
+                                    .map((item) => InterestCard(
+                                          interest: item,
+                                          width: anchorSize * 0.30,
+                                          isSelected: state.selectedInterests
+                                              .contains(item),
+                                          onPressed: () {
+                                            context
+                                                .read<InterestsCubit>()
+                                                .selectInterest(item);
+                                          },
+                                        ))
+                                    .toList(),
+                              ),
+                            ),
+                          SizedBox(
+                            height: size.height * 0.10,
+                          ),
+                        ],
+                      ),
+                    ),
+                  ),
+                  Positioned(
+                    top: 30,
+                    left: 30,
+                    child: givt_widgets.BackButton(
+                      pageName: Pages.interestsSelection.name,
+                    ),
+                  ),
+                  Positioned(
+                    top: 30,
+                    right: 30,
+                    child: ProfileWalletButton(
+                      pageName: Pages.interestsSelection.name,
+                    ),
+                  ),
+                ],
+              ),
+              floatingActionButtonLocation:
+                  FloatingActionButtonLocation.centerFloat,
+              floatingActionButton: Container(
+                margin: const EdgeInsets.only(bottom: 25),
+                child: GivtPrimaryElevatedButton(
+                  padding:
+                      const EdgeInsets.symmetric(horizontal: 100, vertical: 12),
+                  text: 'Next',
+                  onPressed: state.selectedInterests.length ==
```

Comment:
Maybe the button should have a disabled state? (including style) instead of removing the onPressed?

Reply:
Actually this is how disabled state for buttons works in Flutter - you just put null to onPressed and a button appears in a disabled state :)